index.d.ts

```diff
@@ -19,7 +21,8 @@ declare const detectNewline: {
 
 	@returns Returns detected newline or `\n` when no newline character is found.
 	*/
-	graceful(string: string): '\r\n' | '\n';
+	graceful<T>(string: T & IsString<T>): '\r\n' | '\n';
```

Comment:
Why not just?
```suggestion
	graceful(string: string): '\r\n' | '\n';
```

Reply:
I swear it, I tried that but `index.test-d.ts` was popping up with errors when supplying `null` or `undefined` unless I did that extra-complicated/redundant `IsString` check.

Somehow those errors have *randomly* disappeared. :smile:

Reply:
Fixed in a0bb6ac2f4377a053f7974180ecc6294f86cade5